### PR TITLE
Add SQS for active job

### PR DESCRIPTION
**Pull request by sharmaansh21**
This is still work in progress. But before i complete it i wanna know what others think about it.

### CODE REVIEW

File: activejob/lib/active_job/queue_adapters/sqs_adapter.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    #
+    # Read more about SQS {here}[https://aws.amazon.com/sqs/].
+    #
+    # To use SQS set the queue_adapter config to +:sqs+.
+    #
+    #   Rails.application.config.active_job.queue_adapter = :sqs
+    class SqsAdapter
+      def enqueue(job) #:nodoc:
+        JobWrapper.queue_as job.queue_name
+        ActiveSupport::JSON.encode(job.serialize)
```
Comment (seuros) [inline activejob/lib/active_job/queue_adapters/sqs_adapter.rb]: What this does ?

Comment (justanshulsharma) [inline activejob/lib/active_job/queue_adapters/sqs_adapter.rb]: @seuros It's just some testing i was doing, but before i go ahead and complete this feature, i wanna know what others think about having Amazon SQS as adapter.

Comment (justanshulsharma) [inline activejob/lib/active_job/queue_adapters/sqs_adapter.rb]: @seuros yup sort of.

File: Gemfile.lock [hunk 2]
```diff
@@ -202,6 +208,10 @@ GEM
     mysql2 (0.3.19)
     nokogiri (1.6.6.2)
       mini_portile (~> 0.6.0)
+    nokogiri (1.6.6.2-x64-mingw32)
```
Comment (seuros) [inline Gemfile.lock]: why this was added ?

**Comment by seuros** [comment]
If the adapter is 100% compatible with the others adapters, i'm :+1:  to add it to Rails 5.0. 

ping @dhh

**Comment by justanshulsharma** [comment]
Yes it is 100% compatible with the other adapters.

File: activejob/lib/active_job/queue_adapters/shoryuken_adapter.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+          queue = Shoryuken::Client.queues(job.queue_name)
+          queue.send_message(message(job))
+        end
+
+        def enqueue_at(job, timestamp) #:nodoc:
+          register_worker!(job)
+
+          delay = (timestamp - Time.current.to_f).round
+          raise 'The maximum allowed delay is 15 minutes' if delay > 15.minutes
```
Comment (egilburg) [inline activejob/lib/active_job/queue_adapters/shoryuken_adapter.rb]: could raise `ArgumentError`

Comment (egilburg) [inline activejob/lib/active_job/queue_adapters/shoryuken_adapter.rb]: also, what sets this delay?

Comment (meinac) [inline activejob/lib/active_job/queue_adapters/shoryuken_adapter.rb]: Also this should be documented like [this](https://github.com/rails/rails/blob/master/activejob/lib/active_job/queue_adapters.rb#L56).

**Comment by rafaelfranca** [comment]
I remember I was against adding a lot of adapters inside Rails when we were discussion about Active Job. For me it should be the library authors job to maintain their Active Job adapters. I'm fine with adding some adapters by default but this is already causing maintenance burden since Active Job tests is always failing because something on the libraries.

I really believe we should stop including new adapters and encourage library author to maintain their adapters.

@dhh @jeremy what do you think?

**Comment by kuldeepaggarwal** [comment]
Yes, this is also a great idea that we have different adapters with different authors to maintain it, it will distribute the responsibility. :smile:

**Comment by rafaelfranca** [comment]
See https://github.com/rails/rails/issues/23311

**Pull request closed without merging**